src/ucp/core/ucp_mm.c

```diff
@@ -292,3 +292,53 @@ ucs_status_t ucp_mem_unmap(ucp_context_h context, ucp_mem_h memh)
     UCP_THREAD_CS_EXIT(&context->mt_lock);
     return status;
 }
+
+static unsigned ucp_madvise2uct(unsigned flags) 
+{
+    return flags;
+}
+
+ucs_status_t 
+ucp_mem_advise(ucp_context_h context, ucp_mem_h memh, 
+               ucp_mem_map_params_t *params)
+{
+    ucs_status_t status, tmp_status;
+    int md_index;
+
+    if (!(params->field_mask & UCP_MEM_MAP_PARAM_FIELD_LENGTH) ||
+        !(params->field_mask & UCP_MEM_MAP_PARAM_FIELD_ADDRESS) ||
+        !(params->field_mask & UCP_MEM_MAP_PARAM_FIELD_FLAGS)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if ((params->address < memh->address) ||
+        (params->address + params->length > memh->address + memh->length)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    ucs_debug("advice buffer %p length %llu memh %p flags %x",
+               params->address, (unsigned long long)params->length, memh,
+               params->flags);
+
+    if (memh == &ucp_mem_dummy_handle) {
+        return UCS_OK;
+    }
+
+    UCP_THREAD_CS_ENTER(&context->mt_lock);
+
+    status = UCS_OK;
```

Comment:
need to check params->flags before starting the loop over MDs, and figure out the uct_flags according to it, without assuming it's the same numeric value.